identity/src/main/java/com/android/identity/CborUtil.java

```diff
@@ -0,0 +1,305 @@
+package com.android.identity;
+
+import static java.lang.annotation.RetentionPolicy.SOURCE;
+
+import androidx.annotation.IntDef;
+import androidx.annotation.NonNull;
+
+import java.io.ByteArrayInputStream;
+import java.lang.annotation.Retention;
+import java.text.DecimalFormat;
+import java.text.DecimalFormatSymbols;
+import java.util.Collection;
+import java.util.List;
+import java.util.Locale;
+
+import co.nstant.in.cbor.CborDecoder;
+import co.nstant.in.cbor.CborException;
+import co.nstant.in.cbor.model.AbstractFloat;
+import co.nstant.in.cbor.model.ByteString;
+import co.nstant.in.cbor.model.DataItem;
+import co.nstant.in.cbor.model.DoublePrecisionFloat;
+import co.nstant.in.cbor.model.NegativeInteger;
+import co.nstant.in.cbor.model.SimpleValue;
+import co.nstant.in.cbor.model.UnsignedInteger;
+
+class CborUtil {
+    private static final String TAG = "CborUtil";
+
+
+    static final int DIAGNOSTICS_FLAG_EMBEDDED_CBOR = (1 << 0);
+
+    static final int DIAGNOSTICS_FLAG_PRETTY_PRINT = (1 << 1);
+
+    @Retention(SOURCE)
+    @IntDef(
+            flag = true,
+            value = {
+                    DIAGNOSTICS_FLAG_EMBEDDED_CBOR,
+                    DIAGNOSTICS_FLAG_PRETTY_PRINT
+            })
+    public @interface DiagnosticsFlags {
+    }
+
+    // Returns true iff all elements in |items| are not compound (e.g. an array or a map).
+    private static boolean cborAreAllDataItemsNonCompound(@NonNull List<DataItem> items,
+                                                          @DiagnosticsFlags int flags) {
+        for (DataItem item : items) {
+            if ((flags & CborUtil.DIAGNOSTICS_FLAG_EMBEDDED_CBOR) != 0 && item.hasTag()) {
+                return false;
+            }
+            switch (item.getMajorType()) {
+                case ARRAY:
+                case MAP:
+                    return false;
+
+                default:
+                    // Do nothing
+                    break;
+            }
+        }
+        return true;
+    }
+
+    private static boolean cborFitsInSingleLine(@NonNull List<DataItem> items,
+                                                @DiagnosticsFlags int flags) {
+        return cborAreAllDataItemsNonCompound(items, flags) && items.size() < 8;
+    }
+
+    private static void toDiagnostics(@NonNull StringBuilder sb,
+                                      int indent,
+                                      @NonNull DataItem dataItem,
+                                      @DiagnosticsFlags int flags) {
+        int count;
+
+        boolean pretty = ((flags & DIAGNOSTICS_FLAG_PRETTY_PRINT) != 0);
+        String indentString = "";
+        if (pretty) {
+            StringBuilder indentBuilder = new StringBuilder();
```

Comment:
Nit:  If you're building with Java 11 (I think you are?), you can replace this loop with:

`String indentString = " ".repeat(indent);`

Reply:
I don't think we are.